packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -1,22 +1,21 @@
 import {
-    convertColumnToResultsColumn,
-    convertToResultsColumns,
     type PivotChartData,
     type RawResultRow,
     type SemanticLayerPivot,
     type SemanticLayerQuery,
     type VizChartLayout,
     type VizSqlColumn,
 } from '@lightdash/common';
-import { difference } from 'lodash';
 import { ResultsRunner } from '../../../components/DataViz/transformers/ResultsRunner';
 import { apiGetSemanticLayerQueryResults } from '../api/requests';
 
 const transformChartLayoutToSemanticPivot = (
     config: VizChartLayout,
 ): SemanticLayerPivot => {
     return {
-        on: config.x ? [config.x.reference] : [],
+        on: config.x?.reference
+            ? { reference: config.x.reference, type: config.x.type }
+            : undefined,
```

Comment:
hmm. I see... it can only have one on. and its less convenient to have an array. although I think all of three properties `on/index`, `columns/headers`, `values` should ALL be type of array everywhere. including viz (okay maybe not in viz..)